src/document/generator.ml

```diff
@@ -455,6 +455,21 @@ module Make (Syntax : SYNTAX) = struct
             O.documentedSrc (cstr ++ O.txt " " ++ O.keyword "of" ++ O.txt " ")
             @ record fields
 
+    let internal_cstr_arg t =
+      let open Lang.TypeDecl.Constructor in
+      let open Lang.TypeExpr in
+      match t.args with
+      | Tuple type_exprs ->
+          List.exists
+            (fun type_expr ->
+              match type_expr with
+              | Constr (p, _) ->
+                  let open Paths.Path in
+                  is_hidden (p :> Paths.Path.t)
+              | _ -> false)
```

Comment:
What about the other cases that recursively contain an `TypeExpr.t` ? `Alias`, `Arrow`, `Tuple`, `Constr`/`Class` arguments.

For example:
```ocaml
type t = Field of (int * For_generated_code.t) * int (* Tuple *)
type t = Field of For_generated_code.t option (* Constr argument *)
```